os/arch/arm/src/amebasmart/amebasmart_reboot_reason.c

```diff
@@ -112,6 +118,16 @@ static reboot_reason_code_t up_reboot_reason_get_hw_value(void)
 		}
 		/* KM0: IWDG reset*/
 		else if (boot_reason & AON_BIT_RSTF_IWDG) {
+			/* KM0: IWDG is used in KM0 to ensure KM0 is working properly, KM4 is wakeup properly from PG, CA32 power is power-on from PG
+			* backup register will be used to record the case that triggered IWDG reboot
+			*/
+			if (boot_reason_reg2 == 0x2) {
+				BKUP_Write(BKUP_REG2, 0);
+				lldbg("Reboot reason: IWDG reset, KM4 wakeup failed\n");
+			} else if (boot_reason_reg2 == 0x3) {
+				BKUP_Write(BKUP_REG2, 0);
```

Comment:
We initialize the `BKUP_REG1` at the `up_reboot_reason_init`. How about doing this there?

Reply:
BKUP_REG1 is used by Samsung for abort/asset reason code.. That is why we used BKUP_REG2 for specific board reason that happen in KM4 or secure image.